orcid_hub/views.py

```diff
@@ -629,11 +629,13 @@ def action_reset(self, ids):
                         processed_at=None, status=status).where(
                         WorkInvitees.work_record.in_(ids)).execute()
                 elif self.model == AffiliationRecord:
-                    # Delete the userInvitation token when reset to send the mail again.
-                    user_invitation = UserInvitation.get(task_id=task_id)
-                    user_invitation.delete_instance()
-            except UserInvitation.DoesNotExist:
-                pass
+                    # Delete the userInvitation token for selected reset items.
+                    selected_emails = []
+                    for item in self.model.select().where(self.model.id.in_(ids)):
```

Comment:
simple list comprehension would be much more elegant. However that leads to N+N+1 :(
it can be done in a single DELETE query, e.g.:

```python
>>> UserInvitation.delete().where(UserInvitation.email.in_(AffiliationRecord.select(AffiliationRecord.email).where(AffiliationRecord.id.in_([1,2,3]))))
<class 'orcid_hub.models.UserInvitation'> DELETE FROM "user_invitation" WHERE ("email" IN (SELECT "ar"."email" FROM "
affiliation_record" AS ar WHERE ("ar"."id" IN (%s, %s, %s)))) [1, 2, 3]
```